File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    int answer = 0;
    
    // 배열 A와 B를 오름차순으로 정렬
    sort(A.begin(), A.end());
    sort(B.begin(), B.end());
    
    // 배열 A와 B의 각 요소를 순차적으로 곱하여 누적된 값 계산
    for (int i = 0; i < A.size(); ++i) {
        answer += A[i] * B[A.size() - 1 - i]; // B는 내림차순으로 사용
    }
    
    return answer;
}

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.*;
import java.util.stream.*;

class Solution
{
    public int solution(int []A, int []B)
    {
        
        Arrays.sort(A);
        Arrays.sort(B);
                
        int len = A.length;
        int ans = 0;

        for(int i = 0 ; i < len ; i++){
            ans += A[i] * B[len - i - 1];
        }


        
        return ans;
    }
}